api/src/main/java/com/datastrato/gravitino/rel/expressions/Indexes/Index.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+
+package com.datastrato.gravitino.rel.expressions.Indexes;
+
+/** Represents an index on a table. */
```

Comment:
You should add more javadoc to this class and all its interfaces, to tell users what's usage of this class and interfaces.